### PR TITLE
Updated SwitchBot documentation to include details about the lock

### PR DESCRIPTION
**Pull request by dsypniewski**
## Proposed change

Updated SwitchBot documentation to include details about the Lock

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/84673
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/switchbot.markdown [hunk 1]
```diff
@@ -67,7 +68,29 @@ There are three attributes available on the SwitchBot entity to give you more in
 
 - `Retry count`: How many times to retry sending commands to your SwitchBot devices.
 
-### Error codes and troubleshooting
+## SwitchBot Lock
+
+Integration currently only uses the primary lock state, in dual lock mode not all things might work properly.
```
Comment (frenck) [inline source/_integrations/switchbot.markdown]:
```suggestion
The integration currently only uses the primary lock state; in dual lock mode, not all things might work properly.
```

File: source/_integrations/switchbot.markdown [hunk 2]
```diff
@@ -67,7 +68,29 @@ There are three attributes available on the SwitchBot entity to give you more in
...
+
+### Lock Encryption Key
+To be able to use the lock it's encryption key needs to be retrieved from internal SwitchBot API.
+In the `PySwitchbot` Python library there is a [script](https://github.com/Danielhiversen/pySwitchbot/blob/master/scripts/get_encryption_key.py) to make the appropriate calls to that API to retrieve your key.
```
Comment (frenck) [inline source/_integrations/switchbot.markdown]: We should have the users use a python file.

File: source/_integrations/switchbot.markdown [hunk 3]
```diff
@@ -67,7 +68,29 @@ There are three attributes available on the SwitchBot entity to give you more in
...
+You can download the current up-to date version of this script using the following command:
+```shell
+wget https://raw.githubusercontent.com/Danielhiversen/pySwitchbot/master/scripts/get_encryption_key.py
+```
+after downloading the script you can obtain your key details by running it with the following parameters:
+```shell
+python3 get_encryption_key.py MAC_ADDRESS USERNAME
+```
+where `MAC_ADDRESS` is the MAC address of the lock and `USERNAME` is your username used to log in into the SwitchBot app.
+Next you'll be prompted to enter your password, and after that the script should output your `Key ID` and `Encryption key` values required for the lock setup.
```
Comment (frenck) [inline source/_integrations/switchbot.markdown]: How would 95% of our users running a system based on containers do this?

We should not put this burden onto users this way.

If user have to run this:

`python3 get_encryption_key.py MAC_ADDRESS USERNAME`

It means we could have run it for them in our config flows as well.

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by bdraco** [comment]
@dsypniewski https://github.com/home-assistant/core/pull/84830 is merged so you should be able to simplify this now quick a bit.

The lock I ordered was delayed a day so I'll do testing when it arrives in the morning. I think this addresses the concerns, but if anything goes wrong I'll address any quick fixes in a PR, or if its not something I can resolve or it is more involved I'll hide lock support behind a feature flag until it can be fixed.

File: source/_integrations/switchbot.markdown [hunk 4]
```diff
@@ -67,7 +71,44 @@ There are three attributes available on the SwitchBot entity to give you more in
...
+  description: SwitchBot account username
+Password:
+  description: SwitchBot account password
+{% endconfiguration_basic %}
+
+<div class='note warning'>
+This integration doesn't support SSO accounts (Login with Google, etc.) only username and password accounts.
+</div>
+
+### Enter lock encryption key manually
```
Comment (bdraco) [inline source/_integrations/switchbot.markdown]: I think we should move this to the pySwitchbot repo readme and link it so any changes to the procedure are always up to date if we change any of the script params etc

Comment (dsypniewski) [reply source/_integrations/switchbot.markdown]: Pushed an update doing just that, I will make a PR for the library a little bit later.

File: source/_integrations/switchbot.markdown (hunk 3 above)
Comment (dsypniewski) [reply source/_integrations/switchbot.markdown]: I think with the recent changes to the config flow, all your concerns should have been addresses.

File: source/_integrations/switchbot.markdown (hunk 1 above)
Comment (dsypniewski) [reply source/_integrations/switchbot.markdown]: Changed as suggested.

File: source/_integrations/switchbot.markdown [hunk 5]
```diff
@@ -67,7 +71,38 @@ There are three attributes available on the SwitchBot entity to give you more in
...
+This option is for those that would rather obtain the encryption key themselves, and/or want to know exactly where and how are their account credentials used.
+
+{% configuration_basic %}
+Key ID:
+  description: Locks' encryption key ID
+Encryption key:
+  description: Locks' encryption key
+{% endconfiguration_basic %}
+
+For instructions on how to obtain locks encryption key see README in [PySwitchbot](https://github.com/Danielhiversen/pySwitchbot) project.
```
Comment (bdraco) [inline source/_integrations/switchbot.markdown]:
```suggestion
For instructions on how to obtain locks encryption key see README in [PySwitchbot](https://github.com/Danielhiversen/pySwitchbot#obtaining-locks-encryption-key) project.
```

File: source/_integrations/switchbot.markdown [hunk 6]
```diff
@@ -67,7 +71,38 @@ There are three attributes available on the SwitchBot entity to give you more in
...
+This option is for those that would rather obtain the encryption key themselves, and/or want to know exactly where and how are their account credentials used.
+
+{% configuration_basic %}
+Key ID:
+  description: Locks' encryption key ID
+Encryption key:
+  description: Locks' encryption key
+{% endconfiguration_basic %}
+
+For instructions on how to obtain locks encryption key see README in [PySwitchbot](https://github.com/Danielhiversen/pySwitchbot#obtaining-locks-encryption-key) project.
```
Comment (frenck) [inline source/_integrations/switchbot.markdown]:
```suggestion
For instructions on how to obtain the locks encryption key, see README in [PySwitchbot](https://github.com/Danielhiversen/pySwitchbot#obtaining-locks-encryption-key) project.
```

File: source/_integrations/switchbot.markdown [hunk 7]
```diff
@@ -67,7 +71,38 @@ There are three attributes available on the SwitchBot entity to give you more in
...
+  description: SwitchBot account username
+Password:
+  description: SwitchBot account password
+{% endconfiguration_basic %}
+
+<div class='note warning'>
+This integration doesn't support SSO accounts (Login with Google, etc.) only username and password accounts.
+</div>
+
+### Enter lock encryption key manually
```
Comment (frenck) [inline source/_integrations/switchbot.markdown]:
```suggestion
### Enter the lock encryption key manually

```

File: source/_integrations/switchbot.markdown [hunk 8]
```diff
@@ -67,7 +71,38 @@ There are three attributes available on the SwitchBot entity to give you more in
 
 - `Retry count`: How many times to retry sending commands to your SwitchBot devices.
 
-### Error codes and troubleshooting
+## SwitchBot Lock
+
+The integration currently only uses the primary lock state; in dual lock mode, not all things might work properly.
+
+A SwitchBot lock can be set up in Home Assistant in two different ways. You can enter the key id and encryption key yourself, or Home Assistant can import them from your SwitchBot account.
+### SwitchBot account (recommended)
```
Comment (frenck) [inline source/_integrations/switchbot.markdown:80]:
```suggestion

### SwitchBot account (recommended)

```

**Review by frenck — APPROVED** [review]
Thanks!

Happy New Year 🥂 

../Frenck

**Pull request merged**